tensorflow_addons/losses/triplet.py

```diff
@@ -175,3 +227,43 @@ def get_config(self):
         }
         base_config = super().get_config()
         return {**base_config, **config}
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class TripletHardLoss(tf.keras.losses.Loss):
+    """Computes the triplet loss with hard negative and hard positive mining.
+
+    The loss encourages the maximum positive distance (between a pair of embeddings
+    with the same labels) to be smaller than the minimum negative distance plus the
+    margin constant in the mini-batch.
+    The loss selects the hardest positive and the hardest negative samples
+    within the batch when forming the triplets for computing the loss.
+    See: https://arxiv.org/pdf/1703.07737.
+
+    We expect labels `y_true` to be provided as 1-D integer `Tensor` with shape
+    [batch_size] of multi-class integer labels. And embeddings `y_pred` must be
+    2-D float `Tensor` of l2 normalized embedding vectors.
+
+    Args:
+      margin: Float, margin term in the loss definition. Default value is 1.0.
+      soft: Boolean, if set, use the soft margin version. Default value is False.
+      name: Optional name for the op.
+    """
+
+    @typechecked
+    def __init__(
+        self, margin: float = 1.0, soft: bool = False, name: str = None, **kwargs
+    ):
+        super().__init__(name=name, reduction=tf.keras.losses.Reduction.NONE)
+        self.margin = margin
+        self.soft = soft
+
+    def call(self, y_true, y_pred):
+        return triplet_hard_loss(y_true, y_pred, self.margin, self.soft)
+
+    def get_config(self):
+        config = {
+            "margin": self.margin,
```

Comment:
Should we also serialize `self.soft` here?

Reply:
Yes